openedx/core/djangoapps/user_api/management/commands/retire_user.py

```diff
@@ -62,9 +63,14 @@ def check_user_exist(self, user_model, userfile, user_name, useremail):
         users = []
         user = user_name
         email = useremail
-        if userfile and (not user and not email):
+        bulk_user_retirement_config = BulkUserRetirementConfig.current()
+        if userfile and (not user and not email) or not userfile and not user and not email:
             try:
-                userinfo = open(userfile, 'r')
+                if not userfile:
+                    file_handle = bulk_user_retirement_config.csv_file if bulk_user_retirement_config.is_enabled() else None
+                    userinfo = file_handle.open('r')
+                else:
```

Comment:
@NIXKnight believes this else clause is generating the error, which I agree with, so we likely need to change or remove the test.